org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/commons/snippets/SnippetRegistry.java

```diff
@@ -0,0 +1,380 @@
+	 *                            vscode snippet format.
+	 * @param contextDeserializer the GSON context deserializer used to create Java
+	 *                            context.
+	 * @throws IOException
+	 */
+	public void registerSnippets(Reader in, TypeAdapter<? extends ISnippetContext<?>> contextDeserializer)
+			throws IOException {
+		registerSnippets(in, null, contextDeserializer);
+	}
+
+	/**
+	 * Register the snippets from the given JSON stream with a context.
+	 * 
+	 * @param in             the JSON reader which declares snippets with vscode
+	 *                       snippet format.
+	 * @param defaultContext the default context.
+	 * @throws IOException
+	 */
+	public void registerSnippets(Reader in, ISnippetContext<?> defaultContext) throws IOException {
+		registerSnippets(in, defaultContext, null);
+	}
+
+	/**
+	 * Register the snippets from the given JSON stream with a context.
+	 * 
+	 * @param in                  the JSON reader which declares snippets with
+	 *                            vscode snippet format.
+	 * @param defaultContext      the default context.
+	 * @param contextDeserializer the GSON context deserializer used to create Java
+	 *                            context.
+	 * @throws IOException
+	 */
+	public void registerSnippets(Reader in, ISnippetContext<?> defaultContext,
+			TypeAdapter<? extends ISnippetContext<?>> contextDeserializer) throws IOException {
+		JsonReader reader = new JsonReader(in);
+		reader.beginObject();
+		while (reader.hasNext()) {
+			String name = reader.nextName();
+			Snippet snippet = createSnippet(reader, contextDeserializer);
+			if (snippet.getDescription() == null) {
+				snippet.setDescription(name);
+			}
+			if (snippet.getContext() == null) {
+				snippet.setContext(defaultContext);
+			}
+			registerSnippet(snippet);
+		}
+		reader.endObject();
+	}
+
+	private static Snippet createSnippet(JsonReader reader,
+			TypeAdapter<? extends ISnippetContext<?>> contextDeserializer) throws JsonIOException, JsonSyntaxException {
+		GsonBuilder builder = new GsonBuilder();
+		builder.registerTypeAdapter(Snippet.class, new SnippetDeserializer(contextDeserializer));
+		return builder.create().fromJson(reader, Snippet.class);
+	}
+
+	/**
+	 * Returns all snippets.
+	 * 
+	 * @return all snippets.
+	 */
+	public List<Snippet> getSnippets() {
+		return snippets;
+	}
+
+	/**
+	 * Returns the snippet completion items according to the context filter.
+	 * 
+	 * @param replaceRange       the replace range.
+	 * @param lineDelimiter      the line delimiter.
+	 * @param canSupportMarkdown true if markdown is supported to generate
+	 *                           documentation and false otherwise.
+	 * @param contextFilter      the context filter.
+	 * @return the snippet completion items according to the context filter.
+	 */
+	public List<CompletionItem> getCompletionItems(Range replaceRange, String lineDelimiter, boolean canSupportMarkdown,

```

Comment:
I think after #680 and #669 are both merged, we should use `xml.preferences.quoteStyle` to determine whether or not to use `"` or `'` in the insert snippet text. I can create an issue for that.

Reply:
Yes please create an issue.